packages/nimble/R/MCMC_run.R

```diff
@@ -128,7 +124,19 @@ runMCMC <- function(mcmc,
         samplesMatrix <- as.matrix(mcmc$mvSamples)
         if(nburnin > 0) samplesMatrix <- samplesMatrix[-(1:nburnin), , drop = FALSE]
         samplesList[[i]] <- samplesMatrix
-        if(WAIC) WAICvalues[i] <- mcmc$calculateWAIC(nburnin = nburnin)
+    }
+    if(WAIC){
+      if(nchains > 1){
+        samplesPerChain <- dim(samplesList[[1]])[1]
+        posteriorSamplesMatrix <- matrix(0, nrow = samplesPerChain*nchains, ncol = dim(samplesList[[1]])[2])
+        for(i in seq_along(samplesList)){
+          posteriorSamplesMatrix[((i-1)*samplesPerChain + 1):(i*samplesPerChain),] <- samplesList[[i]][,]
+        }
+        colnames(posteriorSamplesMatrix) <- colnames(samplesList[[1]])
+        matrix2mv(posteriorSamplesMatrix, mcmc$mvSamples)  ## transfer all posterior samples into mcmc$mvSamples
+        mcmc$calculateWAIC()
+      }
```

Comment:
@NLMichaud Is there an `else` missing here?  I'm just glancing at it quickly, I could be wrong.

Reply:
Oops nice catch!  The call to `mcmc$calculateWAIC()` within the `if(nchains > 1)` statement was unnecessary.